examples/katex/src/content/docs/index.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Starlight + KaTeX
+description: Example of extending Starlight to support KaTex (LaTeX) math syntax
+---
+
+This project sets up the `remark-math` and `rehype-katex` Markdown extensions to add math syntax support.
+
+## Set-up
+
+### Install dependencies
+
+```sh
+npm install remark-math rehype-katex
+```
+
+### Configure `astro.config.mjs`
+
+Import the Markdown plugins and use them in Astro’s `markdown` config:
+
+```js
+import { defineConfig } from 'astro/config';
+import starlight from '@astrojs/starlight';
+import remarkMath from 'remark-math';
+import rehypeKaTeX from 'rehype-katex';
+
+export default defineConfig({
+  // Configure `remark-math` and `rehype-katex` plugins:
+  markdown: {
+    remarkPlugins: [remarkMath],
+    rehypePlugins: [rehypeKaTeX],
+  },
+
+  integrations: [
+    starlight({
+      // ...
+    }),
+  ],
+});
+```
+
+### Fix styling
+
+In order to properly display math content, you have to add KateX css file to the head of your pages.
+
+This can be achieved by adding a `head`attribute to your Starlight config in `astro.config.mjs`:
+
+```js title = 'astro.config.mjs'
+starlight({
+	title: 'KaTeX Example',
+    // Add KaTeX css.
+    head: [
+      {
+        tag: 'link',
+        attrs: {
+     	    rel: 'stylesheet',
+          'href': 'https://cdn.jsdelivr.net/npm/katex@0.16.25/dist/katex.css',
+          integrity: 'sha512-1jSEPj+0P3loEV9Bqx/3V6MXp4SGDSuLZwvwOECspSHwdHhQvGd27PJba8AVpwUldurvLTgzN+oX37uNiwOw==',
+          crossorigin: 'anonymous'
+      },
+  	},
+  ],
+}),
+```
```

Comment:
This can probably be trimmed down a bit. The example is designed to show the code itself, so no need to repeat it here (otherwise there are two locations to maintain if we need to update it/make changes). Instead it can just be a quick summary mentioning that the Markdown plugins were added and custom CSS too.